packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksEditor.tsx

```diff
@@ -1,50 +1,49 @@
 import * as React from 'react';
 
-import { Box, Flex, Typography, InputWrapper, Divider } from '@strapi/design-system';
+import { createContext } from '@radix-ui/react-context';
+import { InputWrapper, Divider } from '@strapi/design-system';
 import { type Attribute } from '@strapi/types';
-import { type MessageDescriptor, useIntl } from 'react-intl';
-import { type Descendant, type Editor, createEditor } from 'slate';
+import { MessageDescriptor, useIntl } from 'react-intl';
+import { type Editor, type Descendant, createEditor } from 'slate';
 import { withHistory } from 'slate-history';
-import { Slate, withReact, ReactEditor } from 'slate-react';
+import { Slate, withReact, ReactEditor, useSlate } from 'slate-react';
 import styled from 'styled-components';
 
-// @ts-expect-error TODO convert to typescript
-import { Hint } from '../Hint';
-
-import { BlocksInput } from './BlocksInput';
+import { BlocksContent } from './BlocksContent';
+import { BlocksToolbar } from './BlocksToolbar';
 import { withLinks } from './plugins/withLinks';
 import { withStrapiSchema } from './plugins/withStrapiSchema';
-import { BlocksToolbar } from './Toolbar/Toolbar';
 
-const TypographyAsterisk = styled(Typography)`
-  line-height: 0;
-`;
+/* -------------------------------------------------------------------------------------------------
+ * BlocksEditorProvider
+ * -----------------------------------------------------------------------------------------------*/
 
-const LabelAction = styled(Box)`
-  svg path {
-    fill: ${({ theme }) => theme.colors.neutral500};
-  }
-`;
+interface BlocksEditorContextValue {
+  // TODO: context data will go here
+}
 
-const EditorDivider = styled(Divider)`
-  background: ${({ theme }) => theme.colors.neutral200};
-`;
+const [BlocksEditorProvider, usePartialBlocksEditorContext] =
+  createContext<BlocksEditorContextValue>('BlocksEditor');
 
-/**
- * Images are void elements. They handle the rendering of their children instead of Slate.
- * See the Slate documentation for more information:
- * - https://docs.slatejs.org/api/nodes/element#void-vs-not-void
- * - https://docs.slatejs.org/api/nodes/element#rendering-void-elements
- */
-const withImages = (editor: Editor) => {
-  const { isVoid } = editor;
+function useBlocksEditorContext(
+  consumerName: string
+): BlocksEditorContextValue & { editor: Editor } {
+  const context = usePartialBlocksEditorContext(consumerName);
```

Comment:
@joshuaellis do you know why radix needs a consumerName? I'm tempted to make it "useBlocksEditorContext" here to avoid having to give a consumer name every time

Reply:
It needs a name because it adds safety around the idea if you use this hook in Component X & then use Component X outside of the context it will tell you exactly where and what's going on :)

Reply:
@remidej Could you please update why we use radix-context and benefit mentioned below in comment as its not documented anywhere.

> It needs a name because it adds safety around the idea if you use this hook in Component X & then use Component X outside of the context it will tell you exactly where and what's going on :)